alica_engine/src/engine/scheduler/JobQueue.cpp

```diff
@@ -9,40 +11,27 @@ void JobQueue::insert(std::shared_ptr<Job>&& job)
     _queue.insert(std::upper_bound(_queue.begin(), _queue.end(), job), std::move(job));
```

Comment:
why is scheduled time required for a job anymore? The jobs simply need to be executed in the order in which they are added. Infact I don't see any need for the job & job queue classes anymore. All you need is 
```
std::vector<std::function<void()>>
std::unordered_map<int64_t, Timers>
```